packages/framework/src/Model/Order/Processing/OrderProcessorMiddleware/ApplyNominalPromoCodeMiddleware.php

```diff
@@ -0,0 +1,122 @@
+        protected readonly DiscountCalculation $discountCalculation,
+        protected readonly VatFacade $vatFacade,
+        protected readonly PriceExtension $priceExtension,
+    ) {
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessingData $orderProcessingData
+     * @param \Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessingStack $orderProcessingStack
+     * @return \Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessingData
+     */
+    public function handle(
+        OrderProcessingData $orderProcessingData,
+        OrderProcessingStack $orderProcessingStack,
+    ): OrderProcessingData {
+        $appliedPromoCodes = $orderProcessingData->orderInput->getPromoCodes();
+
+        $orderData = $orderProcessingData->orderData;
+
+        foreach ($appliedPromoCodes as $appliedPromoCode) {
+            if ($appliedPromoCode->getDiscountType() !== PromoCode::DISCOUNT_TYPE_NOMINAL) {
+                continue;
+            }
+
+            try {
+                $promoCodeLimit = $this->promoCodeFacade->getHighestLimitByPromoCodeAndTotalPrice($appliedPromoCode, $orderData->totalPricesByItemType[OrderItemTypeEnum::TYPE_PRODUCT]);
+            } catch (PromoCodeException) {
+                continue;
+            }
+
+            $discountOrderItemData = $this->createDiscountOrderItemData(
+                $appliedPromoCode,
+                $promoCodeLimit,
+                $orderProcessingData->getDomainConfig(),
+            );
+
+            if ($discountOrderItemData === null) {
+                continue;
+            }
+
+            $orderData->addItem($discountOrderItemData);
+            $orderData->addTotalPrice($discountOrderItemData->getTotalPrice(), OrderItemTypeEnum::TYPE_DISCOUNT);
+        }
+
+        return $orderProcessingStack->processNext($orderProcessingData);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCode $promoCode
+     * @param \Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCodeLimit\PromoCodeLimit $promoCodeLimit
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig $domainConfig
+     * @return \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData|null
+     */
+    protected function createDiscountOrderItemData(
+        PromoCode $promoCode,
+        PromoCodeLimit $promoCodeLimit,
+        DomainConfig $domainConfig,
+    ): ?OrderItemData {
+        $locale = $domainConfig->getLocale();
+        $domainId = $domainConfig->getId();
+
+        $defaultVat = $this->vatFacade->getDefaultVatForDomain($domainId);
+
+        $discountPrice = $this->discountCalculation->calculateNominalDiscount(
+            Money::create($promoCodeLimit->getDiscount()),
+            (float)$defaultVat->getPercent(),
+        );
+
+        $discountOrderItemData = $this->orderItemDataFactory->create(OrderItemTypeEnum::TYPE_DISCOUNT);
+
+        $discountPrice = $discountPrice->inverse();
```

Comment:
thanks to this line, the `$discountPrice` is negative, right? If so, you should remove the `-` sign from the sprintf call a few lines below